### PR TITLE
KL Divergence Early Stopping

### PR DESCRIPTION
**Pull request by martindevans**
Added an early stopping policy to training which prevents overtraining on one particular set of data. if the model output changes a lot within a single generation then that generation training is ended early.

This should stabilise training, preventing huge spikes and then dips in reward (at the cost of adding another hyperparameter)

### CODE REVIEW

File: rlpy/training.py [hunk 1]
```diff
@@ -208,6 +213,17 @@ def train(model, dataset_in: pd.DataFrame, dataset_out: pd.DataFrame, dataset_ex
                  + parameters.large_entropy_coeff * std_penalty
             )
 
+            # Calculate an approximation of the KL divergence between the old and new policies
+            log_ratio = (new_log_probs - batch_old_log_probs)
+            kl_divergence = torch.mean((ratio - 1) - log_ratio)
```
[the remark this answers is not in the document]
Comment (martindevans) [reply rlpy/training.py]: Incorrect, see referenced code

**Review by 1Solon — APPROVED** [review]
